Mandiant v 1 1

adamlevymandiant opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
NA

## Description
New version that:

- Splits Feed and Enrichment commands into separate integrations
- Enhances the readable output for the `cve`, `mati-get-actor`, `mati-get-malware`, `mati-get-campaign` commands

## Must have
- [x] Tests
- [x] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=36047) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=36047) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=36047) it.</sub>

kgal-pan commented:
@adamlevymandiant - Can you please sync the fork? It should resolve the [Setup Environment check failing](https://app.circleci.com/pipelines/github/demisto/content/379113/workflows/46eda44e-3d05-429b-83cf-e273ec14452a/jobs/751431) and the [pre-commit validation failure](https://github.com/demisto/content/actions/runs/10583280098/job/29354583437?pr=36047#step:8:831).


Also, can you please increase the code coverage for Feed Mandiant integration? Seems that we're [lower than the 70% threshold](https://github.com/demisto/content/actions/runs/10583280098/job/29354583437?pr=36047#step:8:1027).

adamlevymandiant commented:
@kgal-pan - thank you for the guidance, I have synced my fork and pushed a commit increasing test coverage above 70%

barryyosi-panw commented:
Hi @adamlevymandiant, since you were splitting the original integration into two separate ones, it still makes sense to use a shared client module for all the obvious reasons.
I'd suggest you to try creating a shared client module, and import it into the two integrations, let me know if you wish to try it and need assistance with that.

In addition, I see you've got some validations failing - try using `demisto-sdk format -i Packs/MandiantAdvantageThreatIntelligence/Integrations/Mandiant/Mandiant.yml` and 
`demisto-sdk format -i Packs/MandiantAdvantageThreatIntelligence/Integrations/FeedMandiant/FeedMandiant.yml`  

it should solve the formatting problems.

adamlevymandiant commented:
@barryyosi-panw - thank you for the guidance, is there a pack that already uses a shared client module and custom docker image that I can use as reference?

barryyosi-panw commented:
Surprisingly, it's actually a precedence, and we don't have a reference of such mechanism. 
Lets keep the current solution for now @adamlevymandiant, your PR will be reviewed shortly.

kgal-pan commented on Packs/MandiantAdvantageThreatIntelligence/Integrations/FeedMandiant/FeedMandiant.yml:
```diff
@@ -0,0 +1,134 @@
+  type: 8
+script:
+  commands:
+  - name: mandiant-get-indicators
+    arguments:
+    - description: The maximum number of indicators to fetch.
+      name: limit
```
Please add a `description` key to the `mandiant-get-indicators` command to fix the [pre-commit validation failure](https://github.com/demisto/content/actions/runs/10697530257/job/29654917907?pr=36047#step:8:875).

```suggestion
  - name: mandiant-get-indicators
    arguments:
    - description: The maximum number of indicators to fetch.
      name: limit
    description: Fetch indicators.
```

kgal-pan replied:
Pushed the change myself in https://github.com/demisto/content/pull/36047/commits/f3f3aba0911da1084edd1fe967857079d451cd6d.

adamlevymandiant commented:
Thanks @kgal-pan - I see you have pushed some commits, do you need me to do anything at this stage?

kgal-pan commented:
> Thanks @kgal-pan - I see you have pushed some commits, do you need me to do anything at this stage?

Sure thing @adamlevymandiant - I just pushed a commit to fix the graph/neo4j crashing in the pre-commit validation stage because of the missing `description` field for the command.

Now that this commit fixed the graph crash, it exposed some things that need to be fixed in the integration. Please review the listed errors starting from [here](https://github.com/demisto/content/actions/runs/10702193210/job/29669821546?pr=36047#step:8:834).

> Is it too late to change the email in `pack_metadata.json` to [mandiant-support@google.com](mailto:mandiant-support@google.com)?

No, feel free to change this.

adamlevymandiant commented:
@kgal-pan - FYI that circleci checks are now passing. Please let me know if anything else is required from my side.

ShirleyDenkberg commented on Packs/MandiantAdvantageThreatIntelligence/ReleaseNotes/1_1_0.md:
```diff
@@ -0,0 +1,8 @@
+#### Integrations
+##### Mandiant Enrich
+- New Integration providing generic enrichment commands.
+- Includes enhancements to the readable output for the `cve`, `mati-get-actor`, `mati-get-malware`, `mati-get-campaign` commands.
+##### Mandiant Feed
+- New Integration providing an indicator feed.
```
```suggestion
- New integration providing generic enrichment commands.
- Includes enhancements to the readable output for the following commands:
   -  `cve`
   - `mati-get-actor`
   - `mati-get-malware`
   - `mati-get-campaign`
##### Mandiant Feed
New integration providing an indicator feed.
```

ShirleyDenkberg commented on Packs/MandiantAdvantageThreatIntelligence/ReleaseNotes/1_1_0.md:
```diff
@@ -0,0 +1,8 @@
+##### Mandiant Feed
+- New Integration providing an indicator feed.
+##### Mandiant Advantage Threat Intelligence
+- This pack now includes individual Feed and Enrichment integrations. Therefore this integration is now deprecated. Please follow the stpes in the documentation to update your integration with Mandiant.
```
```suggestion
This pack now includes individual Feed and Enrichment integrations. Therefore this integration is now deprecated. Please follow the steps in the documentation to update your integration with Mandiant.
```

ShirleyDenkberg commented:
@kgal-pan @barryyosi-panw @MLainer1 Doc review completed.

thefrieddan1 commented:
> @kgal-pan - FYI that circleci checks are now passing. Please let me know if anything else is required from my side.

Great. There are now comments from our technical writing expert.
Our next step will be to setup a demo or you can send me a recording of the new features you have added.

adamlevymandiant commented:
Happy to set up a demo if you want to let me know a few times that work for you?

Should I send a new commit addressing the feedback from the tech writing expert? Will that trigger a whole set of new reviews if I do that?
